zerver/migrations/0517_set_subscribe_to_default_streams_for_existing_objects.py

```diff
@@ -0,0 +1,36 @@
+# Generated by Django 5.0.5 on 2024-04-30 09:22
+
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+
+def set_subscribe_to_default_streams_for_existing_objects(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    PreregistrationUser = apps.get_model("zerver", "PreregistrationUser")
+    MultiuseInvite = apps.get_model("zerver", "MultiuseInvite")
+
+    PreregistrationUser.objects.filter(
+        subscribe_to_default_streams=None, referred_by=None, multiuse_invite=None, streams=None
+    ).update(subscribe_to_default_streams=True)
+    PreregistrationUser.objects.filter(subscribe_to_default_streams=None).update(
+        subscribe_to_default_streams=False
+    )
+    MultiuseInvite.objects.filter(subscribe_to_default_streams=None).update(
+        subscribe_to_default_streams=False
```

Comment:
The invitations table is quite large, at the scale of users, but it's not exactly a hot table, so it might be fine to do such a small write to it in a single batch. Would be safer to just batch it and there's no harm in doing so.